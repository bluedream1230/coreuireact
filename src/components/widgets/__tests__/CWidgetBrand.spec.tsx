import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { CWidgetBrand } from '../../../index'

test('loads and displays CWidgetBrand component', async () => {
  const { container } = render(<CWidgetBrand/>)
  expect(container).toMatchSnapshot()
})

test('CWidgetBrand customize', async () => {
    const { container } = render(
        <CWidgetBrand
            className="bazinga"
            color="info"
            values={[['aaa', 'bbb'], ['ccc', 'ddd']]}
        />
    )
    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('bazinga')
    if(container.firstChild === null){
        expect(true).toBe(false)
    }else{
        expect(container.firstChild.firstChild).toHaveClass('position-relative')
        expect(container.firstChild.firstChild).toHaveClass('d-flex')
        expect(container.firstChild.firstChild).toHaveClass('justify-content-center')
        expect(container.firstChild.firstChild).toHaveClass('align-items-center')
        expect(container.firstChild.firstChild).toHaveClass('bg-info')
        expect(container.firstChild.lastChild).toHaveClass('row')
        expect(container.firstChild.lastChild).toHaveClass('text-center')
        if(container.firstChild.lastChild === null || container.firstChild.lastChild.firstChild === null){
            expect(true).toBe(false)
        }else{
            expect(container.firstChild.lastChild.firstChild.firstChild).toHaveClass('fs-5')
            expect(container.firstChild.lastChild.firstChild.firstChild).toHaveClass('fw-semibold')
            expect(container.firstChild.lastChild.firstChild.firstChild).toHaveTextContent('aaa')
            expect(container.firstChild.lastChild.firstChild.lastChild).toHaveClass('text-uppercase')
            expect(container.firstChild.lastChild.firstChild.lastChild).toHaveClass('text-medium-emphasis')
            expect(container.firstChild.lastChild.firstChild.lastChild).toHaveClass('small')
            expect(container.firstChild.lastChild.firstChild.lastChild).toHaveTextContent('bbb')
        }
    }

})