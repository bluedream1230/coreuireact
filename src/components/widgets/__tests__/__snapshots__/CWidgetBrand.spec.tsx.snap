// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CWidgetBrand customize 1`] = `
<div>
  <div
    class="card bazinga"
  >
    <div
      class="card-header position-relative d-flex justify-content-center align-items-center bg-info"
    />
    <div
      class="card-body row text-center"
    >
      <div
        class="col"
      >
        <div
          class="fs-5 fw-semibold"
        >
          aaa
        </div>
        <div
          class="text-uppercase text-medium-emphasis small"
        >
          bbb
        </div>
      </div>
      <div
        class="vr"
      />
      <div
        class="col"
      >
        <div
          class="fs-5 fw-semibold"
        >
          ccc
        </div>
        <div
          class="text-uppercase text-medium-emphasis small"
        >
          ddd
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`loads and displays CWidgetBrand component 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-header position-relative d-flex justify-content-center align-items-center"
    />
    <div
      class="card-body row text-center"
    />
  </div>
</div>
`;
